clouddriver-event/src/main/kotlin/com/netflix/spinnaker/clouddriver/event/persistence/EventRepository.kt

```diff
@@ -49,8 +51,30 @@ interface EventRepository {
   /**
    * List all aggregates for a given type.
    *
-   * @param aggregateType The aggregate collection name. If not provided, all aggregates will be returned.
-   * @return An unordered list of matching aggregates.
+   * @param criteria The criteria to limit the response by
+   * @return A list of matching aggregates
    */
-  fun listAggregates(aggregateType: String?): List<Aggregate>
+  fun listAggregates(criteria: ListAggregatesCriteria): ListAggregatesResponse
+
+  /**
+   * @param aggregateType The type of [Aggregate] to return. If unset, all types will be returned.
+   * @param token The page token to paginate from. It will return the first results
+   * @param perPage The number of [Aggregate]s to return in each response
+   */
+  class ListAggregatesCriteria(
+    val aggregateType: String? = null,
+    val token: String? = null,
+
+    @Positive @Max(1000)
+    val perPage: Int = 100
+  )
+
+  /**
```

Comment:
Is this representing Response for a Aggregate request ?

Reply:
It's representing the response from the `listAggregates` method. Not to be confused with an HTTP response or something else. Would it make more sense if it were renamed ot `ListAggregatesResult`?

Reply:
+1 for ListAggregatesResult